e2etest/bgp_test.go

```diff
@@ -687,12 +689,76 @@ var _ = ginkgo.Describe("BGP", func() {
 				pairExternalFRRWithNodes(cs, c, ipFamily)
 
 				for j := 0; j <= i; j++ {
-					validateFRRPeeredWithNodes(cs, frrContainers[j], ipFamily)
+					validateFRRPeeredWithNodes(cs, frrContainers[j], ipFamily, BeNil)
 				}
 			}
 		},
 			table.Entry("IPV4", "ipv4"),
 			table.Entry("IPV6", "ipv6"))
+
+		table.DescribeTable("configure bgp community NOPEER attribute and verify FRR session can't get to established state",
+			func(addressPools []addressPool, chkFunc func() types.GomegaMatcher, ipFamily string) {
+				configData := configFile{
+					Peers: peersForContainers(frrContainers, ipFamily),
+					Pools: addressPools,
+				}
+				for _, c := range frrContainers {
+					pairExternalFRRWithNodes(cs, c, ipFamily)
+				}
+
+				err := updateConfigMap(cs, configData)
+				framework.ExpectNoError(err)
+
+				for _, c := range frrContainers {
+					validateFRRPeeredWithNodes(cs, c, ipFamily, chkFunc)
+				}
+			},
+			table.Entry("IPV4", []addressPool{
+				{
+					Name:     "bgp-test",
+					Protocol: BGP,
+					Addresses: []string{
+						"192.168.10.0/24",
+					},
+					BGPAdvertisements: []bgpAdvertisement{
+						{
+							Communities: []string{
+								CommunityNOPeer,
```

Comment:
This should contain the name of a community defined in the communities section of the configuration, not the community itself.

Reply:
it can be the name or community itself both are supported

Reply:
ah, neat!